### PR TITLE
Add specs for Enumerable#min with optional arg. 🐒

### PR DESCRIPTION
**Pull request by jsyeo**
:smile: Second PR for #3264's checklist. Along with some minor cleanup of min's specs.

### CODE REVIEW

File: spec/ruby/core/enumerable/min_spec.rb [hunk 1]
```diff
@@ -88,4 +86,21 @@
     multi.min.should == [1, 2]
   end
 
+  context "when called with an argument n" do
+    context "without a block" do
+      it "returns an array containing the minimum n elements" do
+        result = @e_ints.min(2)
+        result.should == [22, 333]
+        result.should be_an_instance_of(Array)
```
Comment (YorickPeterse) [inline spec/ruby/core/enumerable/min_spec.rb]: This seems a little bit redundant since the above line already compares `result` with an Array.

Comment (jsyeo) [inline spec/ruby/core/enumerable/min_spec.rb]: Oops. Let me fix that.

**Comment by jsyeo** [comment]
@YorickPeterse Removed redundant expectation.

**Comment by YorickPeterse** [comment]
Thanks! :+1:

**Pull request merged**